libs/blocks/countdown-timer/countdown-timer.js

```diff
@@ -0,0 +1,240 @@
+      const endTime = this.timeRangesEpoch[i + 1];
+
+      if (currentTime >= startTime && currentTime <= endTime) {
+        this.isVisible = true;
+        const diffTime = endTime - currentTime;
+        this.daysLeft = Math.floor(diffTime / (1000 * 60 * 60 * 24));
+        this.hoursLeft = Math.floor((diffTime % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+        this.minutesLeft = Math.floor((diffTime % (1000 * 60 * 60)) / (1000 * 60));
+        this.update();
+        return;
+      }
+    }
+    this.countdownCompleted();
+  }
+
+  update() {
+    if (!this.isVisible) {
+      this.shadowRoot.innerHTML = '';
+      return;
+    }
+
+    this.shadowRoot.innerHTML = `
+      <style>${CountdownTimer.css}</style>
+      <div class="${this.classList}">
+        <div class="timer-label">${this.label}</div>
+        <div class="timer-block">
+          <div id="cdt-days" class="timer-fragment">
+              <div class="timer-unit-container">
+                <div class="timer-box">${Math.floor(this.daysLeft / 10)}</div>
+                <div class="timer-box">${this.daysLeft % 10}</div>
+              </div>
+              <div class="timer-unit-label">${this.timeWords[0]}</div>
+          </div>
+          <div class="timer-label">:</div>
+          <div id="cdt-hours" class="timer-fragment">
+              <div class="timer-unit-container">
+                <div class="timer-box">${Math.floor(this.hoursLeft / 10)}</div>
+                <div class="timer-box">${this.hoursLeft % 10}</div>
+              </div>
+              <div class="timer-unit-label">${this.timeWords[1]}</div>
+          </div>
+          <div class="timer-label">:</div>
+          <div id="cdt-minutes" class="timer-fragment">
+              <div class="timer-unit-container">
+                <div class="timer-box">${Math.floor(this.minutesLeft / 10)}</div>
+                <div class="timer-box">${this.minutesLeft % 10}</div>
+              </div>
+              <div class="timer-unit-label">${this.timeWords[2]}</div>
+          </div>
+        </div>
+      </div>
+    `;
+  }
+}
+
+customElements.define('countdown-timer', CountdownTimer);
+
+export default function init(el) {
+  const styles = [...el.classList];
+  const firstLevelDivs = el.querySelectorAll(':scope > div');
+
+  // Extract 'DAYS HOURS MINS' from the first div's first child
+  const daysHoursMins = firstLevelDivs[0].querySelector(':scope > div').textContent.trim();
+
+  // Extract 'ENDS IN' from the first div's second child
+  const cdtLabel = firstLevelDivs[0].querySelector(':scope > div:nth-child(2)').textContent.trim();
+
+  // Extract the time ranges from the second and third divs
+  const timeRanges = Array.from(firstLevelDivs)
+    .slice(1) // Skip the first div, as it's not part of the time ranges
+    .flatMap((div) => Array.from(div.querySelectorAll(':scope > div')).map((innerDiv) => Date.parse(innerDiv.textContent.trim()))) // Extract the text content of each inner div
```

Comment:
we need to catch the exception in `Date.parse` just in case it fails in some locales.

Reply:
Done